symbology/custom-dictionary-style/src/main/java/com/esri/samples/custom_dictionary_style/CustomDictionaryStyleSample.java

```diff
@@ -67,31 +92,120 @@ public void start(Stage stage) {
       // set the initial viewpoint to the Esri Redlands campus
       map.setInitialViewpoint(new Viewpoint(new Point(-1.304630524635E7, 4036698.1412000023, map.getSpatialReference()), 5000));
 
-      // create the restaurants feature table from the feature service
-      ServiceFeatureTable restaurantFeatureTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/arcgis/rest/services/Redlands_Restaurants/FeatureServer/0");
-
-      // create the restaurants layer and add it to the map
-      FeatureLayer restaurantLayer = new FeatureLayer(restaurantFeatureTable);
-      map.getOperationalLayers().add(restaurantLayer);
-
-      // open the custom style file
-      String styleFilePath = new File(System.getProperty("data.dir"), "./samples-data/stylx/Restaurant.stylx").getAbsolutePath();
-      DictionarySymbolStyle restaurantStyle = DictionarySymbolStyle.createFromFile(styleFilePath);
-
-      // create a dictionary renderer
-      DictionaryRenderer dictionaryRenderer = new DictionaryRenderer(restaurantStyle);
-
-      // set the dictionary renderer to the restaurant layer
-      restaurantLayer.setRenderer(dictionaryRenderer);
-
-      // add the map view and control panel to stack pane
-      stackPane.getChildren().add(mapView);
+      // set up the UI
+      setupUI();
+
+      // enable UI interactions once the feature layer has loaded
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          controlsVBox.setDisable(false);
+        } else {
+          new Alert(Alert.AlertType.ERROR, "Map failed to load").show();
+        }
+      });
+
+      // create a portal item using the portal and the item id of the dictionary web style
+      Portal portal = new Portal("https://arcgisruntime.maps.arcgis.com");
+      PortalItem portalItem = new PortalItem(portal, "adee951477014ec68d7cf0ea0579c800");
+
+      // create a stylex file from a local location
+      File stylxFile = new File(System.getProperty("data.dir"), "./samples-data/stylx/Restaurant.stylx");
+
+      // create a service feature table
+      ServiceFeatureTable serviceFeatureTable = new ServiceFeatureTable("https://services2.arcgis.com/ZQgQTuoyBrtmoGdP/ArcGIS/rest/services/Redlands_Restaurants/FeatureServer/0");
+      // create the restaurants feature layer using the service feature table
+      FeatureLayer restaurantLayer = new FeatureLayer(serviceFeatureTable);
+
+      // map the input fields in the feature layer to the dictionary symbol style's expected fields for symbols and text
+      Map<String, String> fieldMap = new HashMap<>();
```

Comment:
I'm a bit rusty on my Java these days, but I think `Map` is an interface, which would be somewhat ambiguous to put an object of the type `HashMap` in it. Perhaps a better way to write this would be 

`HashMap<String, String> fieldMap = new HashMap<>();`

The method call on line 139 (`new DictionaryRenderer(dictionarySymbolStyle, fieldMap, fieldMap);`) which takes this variable in has a signature that takes a `Map` object, but since `HashMap` is a type of `Map`, it should be all good.

I'd happily stand corrected though!